internal/services/domain/domains_registration.go

```diff
@@ -0,0 +1,676 @@
+			Delete:  schema.DefaultTimeout(defaultDomainRegistrationTimeout),
+			Default: schema.DefaultTimeout(defaultDomainRegistrationTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"domain_names": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "List of domain names to be managed.",
+			},
+			"duration_in_years": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"project_id": account.ProjectIDSchema(),
+
+			"owner_contact_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ExactlyOneOf: []string{
+					"owner_contact_id",
+					"owner_contact",
+				},
+				Description: "ID of the owner contact. Either `owner_contact_id` or `owner_contact` must be provided.",
+			},
+			"owner_contact": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				ExactlyOneOf: []string{
+					"owner_contact_id",
+					"owner_contact",
+				},
+				Elem: &schema.Resource{
+					Schema: contactSchema(),
+				},
+				Description: "Details of the owner contact. Either `owner_contact_id` or `owner_contact` must be provided.",
+			},
+			"administrative_contact": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: contactSchema(),
+				},
+				Description: "Details of the administrative contact.",
+			},
+			"technical_contact": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: contactSchema(),
+				},
+				Description: "Details of the technical contact.",
+			},
+			"auto_renew": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Enable or disable auto-renewal of the domain.",
+			},
+			"dnssec": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Enable or disable dnssec for the domain.",
+			},
+			"ds_record": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key_id": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							Description: "The identifier for the dnssec key.",
+						},
+						"algorithm": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The algorithm used for dnssec (e.g., rsasha256, ecdsap256sha256).",
+						},
+						"digest": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The type of digest (e.g., sha_1, sha_256).",
```

Comment:
Are permitted values in an enum somewhere? If yes it could be helpful to add a validation here with the enum values.